backend/app/Http/Controllers/QuizLogController.php

```diff
@@ -34,6 +35,11 @@ protected function getUserWithQuizLogs($id) {
     }
 
     protected function validateQuizLog(Request $request) {
+        $checkUnique = QuizLog::where('quiz_id', $request->quiz_id)->where('user_id', $request->user_id)->first();
```

Comment:
This can be shorten, you can group `where()` in laravel.
related topic: https://laravel.com/docs/7.x/queries#where-clauses
```suggestion
        $checkDuplicate = Model::where([
            ['column_1', '=', $data['column_1]],
            ['column_2', '=', $data['column_2]],
        ])->get();
```